test/puppeteer_har_server.js

```diff
@@ -0,0 +1,113 @@
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const puppeteer = require("puppeteer");
+const express = require("express");
+const { PuppeteerHARServer } = require("../lib/puppeteer_har_server");
+const {
+  PuppeteerHarServerTestClient
+} = require("./puppeteer_har_server_test_client");
+
+chai.use(chaiHttp);
+chai.should();
+
+const PUPPETEER_HAR_SERVER_PORT = 3000;
+
+function startServer(sessionManager) {
+  const harServer = new PuppeteerHARServer({
+    sessionManager,
+    port: PUPPETEER_HAR_SERVER_PORT
+  });
+  return harServer.start();
+}
+
+function stopServer(server) {
+  return new Promise(resolve => server.close(resolve));
+}
+
+describe("PuppeteerHARServer", () => {
+  describe("POST /har-sessions", () => {
+    it("should return 400 browserURL is unspecified", async () => {
+      const server = startServer();
+      const client = new PuppeteerHarServerTestClient(server);
+
+      const response = await client.createHarSession(null);
+      response.should.have.status(400);
+      response.body.should.deep.equal({ error: "Missing browserURL" });
+
+      await stopServer(server);
+    });
+
+    it("should return 500 if unexpected error occurs when creating session", async () => {
+      const server = startServer({
+        startSession(browserURL) {
+          throw new Error("Unexpected error");
+        }
+      });
+      const client = new PuppeteerHarServerTestClient(server);
+
+      const response = await client.createHarSession("http://localhost:1337");
+      response.should.have.status(500);
+      response.body.should.deep.equal({});
+
+      await stopServer(server);
+    });
+
+    it("should return session ID as string if session starts successfully", async () => {
```

Comment:
What do you think about making this test name consistent with the corresponding `destroyHarSession` test on line 98?
```suggestion
    it("should return session ID as string if the session was successfully created", async () => {
```

Reply:
Addressed in 3e939380d1dd01d3e3b08f12eb91ad3966478546